synapse/storage/monthly_active_users.py

```diff
@@ -235,8 +235,12 @@ def populate_monthly_active_users(self, user_id):
             # but only update if we have not previously seen the user for
             # LAST_SEEN_GRANULARITY ms
             if last_seen_timestamp is None:
-                count = yield self.get_monthly_active_count()
-                if count < self.hs.config.max_mau_value:
+                # Optimize the db usage when not limiting usage
```

Comment:
I had to look at it a few times to understand the intent - I'm not sure I can suggest a more expressive way to do via code, but a more descriptive comment would be helpful. 

I'm not sure it will necessarily help a great deal with db load since get_monthly_active_count is cached but the logic expects max_mau_value to be set and so I agree that it needs the if check.

Perhaps something like

"In the case where mau_stats_only is True and limit_usage_by_mau is False, there is no point in checking get_monthly_active_count - it adds no value and will break the logic if max_mau_value is exceeded "

Reply:
> I had to look at it a few times to understand the intent 

(I had the same reaction tbh)